### DISCUSSION TITLE
[Feature] - Expose, Store XMP keywords

### DISCUSSION THREAD

**Discussion by psyciknz** [immich-app/immich#10691] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

If a user uses XMP data, specifically categories or events, they seem to be read when refreshing metadata.  But do not look like they are stored for searching.  
Here is a log snippet when a image file and it's XML were refreshed:
```
"Event": "Cyprus Paphos",
  "Subject": [
    "Travel",
    "Cyprus",
    "Paphos",
    "Albums",
    "Cyprus Paphos",
    "2019 World Trip"
  ],
  "HierarchicalSubject": [
    "Travel",
    "Travel|Cyprus",
    "Travel|Cyprus|Paphos",
    "Albums",
    "Albums|2019 World Trip|Cyprus Paphos",
    "Albums|2019 World Trip"
  ],
```
But searching for Albums|2019 World Trip|Cyprus Paphos, Cyprus Paphos which are Categories, or Cyprus Paphos which I've also as an event, seems to perform a city search.

I was hoping ot use this as a method to easily find images that I have categoried.  And then load them into Albums.

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
#3950
